tokens/models.py

```
@@ -28,6 +61,72 @@ def format_price(self, amount_str: str):
+
+        price_data = {}
+        if self.coingecko_id:
+            try:
+                logger.info(
+                    "No existing price within acceptable time period; fetching historical pricefrom gecko..."
+                )
+                endpoint = f"{settings.COINGECKO_URL}/coins/{self.coingecko_id}/history?date={format_date(timestamp)}&localization=false"
+                if settings.COINGECKO_API_KEY:
+                    endpoint += f"&x_cg_pro_api_key={settings.COINGECKO_API_KEY}"
+                logger.info(f"coingecko endpoint: {endpoint}")
+                response = requests.get(endpoint)
```

Ruff (0.15.20)

[error] Probable use of `requests` call without timeout

(S113)